### PR TITLE
fix: handle zero-price listings gracefully in UI and sort

### PR DESCRIPTION
**Pull request by Danielsio**
## Summary

Listings without a price tag (price=0) already flow through the entire pipeline correctly — they're fetched, stored, scored (price dimension omitted, other dimensions reweighted), and notified. This PR fixes the two display-side rough edges:

- **`formatPrice()`** now returns "ללא מחיר" for `price <= 0` instead of rendering "0 ₪"
- **`price_asc` / `price_desc` sort** pushes zero-price listings to the end using a `CASE WHEN price <= 0 THEN 1 ELSE 0 END` prefix, so they don't pollute the top of sorted results

## Test plan

- [x] TypeScript compiles (`tsc --noEmit`)
- [x] Go tests pass (`make test` — 82.2% coverage)
- [ ] Verify listings with price=0 show "ללא מחיר" in the UI
- [ ] Verify sorting by price puts zero-price listings at the bottom

Made with [Cursor](https://cursor.com)

## Release Notes

* **Bug Fixes**
  * Improved sorting of listings with no price in search results—they now appear after priced items while maintaining consistent secondary ordering.
  * Updated price display to show "No price" for listings without a valid price instead of formatting zero or negative values.

* **Chores**
  * Updated project configuration files.

### CODE REVIEW

File: .context/compound-engineering/ce-review/20260504-000806-ca4a117f/correctness.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      "evidence": "consecutiveFailures is a local variable, initialized at the start of Enrich():\n```go\nfunc (e *Enricher) Enrich(ctx context.Context, listings []model.RawListing) (int, error) {\n    enriched := 0\n    attempts := 0\n    consecutiveFailures := 0  // line 51\n```\n\nExit paths:\n- Line 70: `return enriched, nil` (ctx.Done, preserves local state only within this invocation)\n- Line 77: same\n- Line 90: `return enriched, fetcher.ErrChallenge`\n- Line 103: `return enriched, ErrEnrichmentAborted`\n- Line 135: `return enriched, nil`\n\nActually, on closer inspection, consecutiveFailures is a local variable, not a field. It cannot leak across calls. However, the abort threshold is checked mid-loop (line 97), and consecutive failures within the *same* call are counted correctly. The issue is NOT present. This finding should be suppressed.",
+      "suggested_fix": "No fix needed; this is a false positive. consecutiveFailures is a local variable and resets on every Enrich() call."
+    },
```
[the remark this answers is not in the document]
Comment (Danielsio) [reply .context/compound-engineering/ce-review/20260504-000806-ca4a117f/correctness.json]: Removed all `.context/` artifacts from this PR and added `.context/` to `.gitignore`.

File: .context/compound-engineering/ce-review/20260504-000806-ca4a117f/maintainability.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      "location": {
+        "file": "internal/fetcher/yad2/enricher.go",
+        "line": 48,
+        "snippet": "func (e *Enricher) Enrich(ctx context.Context, listings []model.RawListing) int"
+      },
+      "recommendation": "Keep the (int, error) signature. Enrichment stopping early for bot protection is materially different from hitting the budget limit. Return fetcher.ErrChallenge when anti-bot triggers (line 87) and a new ErrEnrichmentAborted when consecutive failures abort (line 100). Let the scheduler decide whether to escalate these conditions (e.g., trigger circuit breaker, notify operators, switch proxy pools). The 3-line error check in scheduler.go is negligible cost compared to the loss of diagnostic signal.",
+      "pre_existing": false
```
[the remark this answers is not in the document]
Comment (Danielsio) [reply .context/compound-engineering/ce-review/20260504-000806-ca4a117f/maintainability.json]: Removed all `.context/` artifacts from this PR and added `.context/` to `.gitignore`.

File: .context/compound-engineering/ce-review/20260504-000806-ca4a117f/project-standards.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      "Commit correctly uses 'Signed-off-by: Daniel Sionov <kingddd301@gmail.com>'",
+      "Commit correctly uses 'Assisted-By: Claude opus 4.6 <noreply@anthropic.com>'",
+      "VIOLATION: Commit incorrectly includes 'Co-authored-by: Cursor <cursoragent@cursor.com>' which is prohibited",
+      "Test coverage maintained at 82.2% (no decrease)",
+      "Tests added for new behavior (defaultMaxPerCycle, isolation, generic error detection)",
+      "Linter passes with no errors (golangci-lint run ./... produced no output)",
+      "All project standards checks passed except prohibited attribution trailer"
+    ]
+  }
+}
```
[the remark this answers is not in the document]
Comment (Danielsio) [reply .context/compound-engineering/ce-review/20260504-000806-ca4a117f/project-standards.json]: Removed all `.context/` artifacts from this PR and added `.context/` to `.gitignore`.

File: .context/compound-engineering/ce-review/20260504-000806-ca4a117f/testing.json [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      "description": "Resource cleanup when constructor fails mid-way",
+      "recommendation": "Add fault injection tests for both NewFetcher and NewFetcherWithProxyPool that fail itemClient creation and verify client cleanup. This is a standard constructor error path pattern."
+    },
+    {
+      "area": "Boundary conditions",
+      "description": "Consecutive failure counter reset at maxConsecutiveFailures boundary",
+      "recommendation": "Add test: 2 failures -> success -> 3 failures should abort after 3 more failures (6 total requests), verifying the counter fully resets to 0 on success, not just below threshold."
+    }
+  ]
+}
```
[the remark this answers is not in the document]
Comment (Danielsio) [reply .context/compound-engineering/ce-review/20260504-000806-ca4a117f/testing.json]: Removed all `.context/` artifacts from this PR and added `.context/` to `.gitignore`.
